R/derive_merged.R

```diff
@@ -0,0 +1,708 @@
+
+  add_data <- filter_if(dataset_add, filter_add) %>%
+    mutate(!!new_var := if_else(!!condition, 1, 0, 0))
+
+  derive_vars_merged(
+    dataset,
+    dataset_add = add_data,
+    by_vars = by_vars,
+    new_vars = vars(!!new_var),
+    order = vars(!!new_var),
+    check_type = "none",
+    mode = "last"
+  ) %>%
+    mutate(!!new_var := if_else(!!new_var == 1, true_value, false_value, false_value))
+}
+
+#' Merge a Character Variable
+#'
+#' Merge a character variable from a dataset to the input dataset. The
+#' observations to merge can be selected by a condition and/or selecting the
+#' first or last observation for each by group.
+#'
+#' @param dataset_add Additional dataset
+#'
+#'   The variables specified by the `by_vars`, the `source_var`, and the `order`
+#'   parameter are expected.
+#'
+#' @param new_var New variable
+#'
+#'   The specified variable is added to the additional dataset and set to the
+#'   transformed value with respect to the `case` parameter.
+#'
+#' @param source_var Source variable
+#'
+#' @param case Change case
+#'
+#'   Changes the case of the values of the new variable.
+#'
+#'   *Permitted Values*: `NULL`, `"lower"`, `"upper"`, `"title"`
+#'
+#' @param missing_value Values used for missing information
+#'
+#'   The new variable is set to the specified value for all by groups without
+#'   observations in the additional dataset.
+#'
+#'   *Permitted Value*: A character scalar
+#'
+#' @inheritParams derive_vars_merged
+#'
+#' @details
+#'
+#'   1. The additional dataset is restricted to the observations matching the
+#'   `filter_add` condition.
+#'
+#'   1. The (transformed) character variable is added to the additional dataset.
+#'
+#'   1. If `order` is specified, for each by group the first or last observation
+#'   (depending on `mode`) is selected.
+#'
+#'   1. The character variable is merged to the input dataset.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @keywords derivation adam
+#'
+#' @export
+#'
+#' @examples
+#' library(admiral.test)
+#' library(dplyr, warn.conflicts = FALSE)
+#' data("dm")
+#' data("ds")
+#'
+#' derive_var_merged_character(
+#'   dm,
+#'   dataset_add = ds,
+#'   by_vars = vars(STUDYID, USUBJID),
+#'   new_var = DISPSTAT,
+#'   filter_add = DSCAT == "DISPOSITION EVENT",
+#'   source_var = DSDECOD,
+#'   case = "title"
+#' ) %>%
+#' select(STUDYID, USUBJID, AGE, AGEU, DISPSTAT)
+derive_var_merged_character <- function(dataset,
+                                        dataset_add,
+                                        by_vars,
+                                        order = NULL,
+                                        new_var,
+                                        source_var,
+                                        case = NULL,
+                                        filter_add = NULL,
+                                        mode = NULL,
+                                        missing_value = NA_character_) {
+
```

Comment:
I understand we created multiple merging functions to avoid users writing expressions in input. But in general for these wrapper functions, I'm wondering is it useful to allow users to keep more source variables (without transformation)? So they don't have to run merging multiple times if they want to merge in multiple variables (for example, from my understanding if they want to merge a converted character variable, but in addition also keep other variables as-is from `dataset_add` they have to run `derive_var_merged_character` then `derive_vars_merged`).

Reply:
You are right, with our current approach the user may need to call `derive_vars_merged_*` functions more than once for merging multiple variables. This was not my favourite solution but the majority preferred this approach. We have discussed it multiple times. Thus I would not like to discuss it again. At least not now. I would wait for user feedback before rediscussing it.

Reply:
I see. Thanks for the information.